### PR TITLE
[OSCD] Add Usage of reg or Powershell by Non-privileged Users rule

### PR DESCRIPTION
**Pull request by ryanplasma**
Page 20 of #574 

This is a bit of a messy one, but I think the end search matches the intention of the slide.

I'm open to suggestions on how to clean it up or write it better.

```
 >tools/sigmac -t es-qs -c helk rules/windows/builtin/win_non_priv_reg_or_ps.yml

((event_id:"1" AND process_integrity_level:"Medium") 
AND ((process_command_line.keyword:*reg* AND process_command_line.keyword:*add*) 
OR (process_command_line.keyword:*powershell* AND process_command_line.keyword:(*set\-itemproperty* OR *\ sp\ * OR *new\-itemproperty*))) 
AND (process_command_line.keyword:*ControlSet* AND process_command_line.keyword:*Services*) 
AND process_command_line.keyword:(*ImagePath* OR *FailureCommand* OR *ServiceDLL*))
```

### CODE REVIEW

**Comment by zinint** [comment]
Greetings, @ryanplasma, thanks for your contribution! 
I'll check the rule and will get back to you.

**Comment by zinint** [comment]
Thanks for using fine `detection` names (instead of `selection#`) ((: 
I think your rule matches the intention of the mentioned slide.
But I also think maybe you could rearrange lines 22-28, e.g.:
  ``` YAML
  powershell:
        CommandLine|contains: 'powershell'
        CommandLine|contains:
            - 'set-itemproperty'
            - ' sp '
            - 'new-itemproperty'
  ```

**Comment by ryanplasma** [comment]
> But I also think maybe you could rearrange lines `22-28`, e.g.:

Yup! I think I had a different idea when I started that section and didn't go back and clean it up.  I'll get that change in ASAP.

**Comment by ryanplasma** [comment]
@zinint looks like duplicate keys make the tests fail.  I can either add the `|all` back in or move it the lines to seperate detections.

File: rules/windows/process_creation/win_non_priv_reg_or_ps.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    reg:
+        CommandLine|contains|all:
+            - reg
+            - add
+    powershell:
+        CommandLine|contains: powershell
+        CommandLine|contains:
+            - set-itemproperty
+            - " sp "
+            - new-itemproperty
```
Comment (yugoslavskiy) [inline rules/windows/process_creation/win_non_priv_reg_or_ps.yml:27]: Hello @ryanplasma !

Thank you for your contribution (: 
Could you please tell me what you've tried to achieve in this section? Maybe I can come up with some solution

Comment (ryanplasma) [reply rules/windows/process_creation/win_non_priv_reg_or_ps.yml:27]: So the idea is that it's looking for ('*powershell*' AND (*set-itemproperty* OR * sp * OR *new-itemproperty*)).  The other idea I had for this was to split it up into `powershell_1` and `powershell_2` and then put the logic in the condition field but there might be a better way so I'm open to any feedback or ideas you have.

Comment (yugoslavskiy) [reply rules/windows/process_creation/win_non_priv_reg_or_ps.yml:27]: I believe that your idea with "`powershell_1` and `powershell_2`" fits best in that scenario (: please update your rule in this way and we will pass your PR for merge.

thank you very much for your work!

Comment (zinint) [reply rules/windows/process_creation/win_non_priv_reg_or_ps.yml:27]: @yugoslavskiy thanks for your help!
@ryanplasma we'll be waiting for you to apply the suggested changes, and I think the PR will be ready (: thanks again for your time and work!

**Comment by yugoslavskiy** [comment]
Thank you for your contribution, @ryanplasma! Great job!

**Review by yugoslavskiy — APPROVED** [review]
@thomaspatzke @Neo23x0 guys, I believe that this PR could be merged (:
please let us know what do you think about it if there is anything you would like to change.

**Pull request merged**